### PR TITLE
DSND-2703: Upgrade to Java 21

### PR DESCRIPTION
**Pull request by sthompsonCH**
This PR upgrades this service to Java 21, upgrades dependencies, and clears out unnecessary dependencies from the pom.

To get Docker working with Testcontainers, we had to exclude the `docker-java-api` dependency from the `data-sync-api-sdk-java` as the docker dep was not up to date and, therefore, incompatible with the most recent version of Spring/Testcontainers. This can be fixed by either explicitly adding the docker dep to the pom with the most recent version, or by excluding it from the offending dependency.

After this, the `TestRestTemplate` was not working when testing a `503` response when Mongo was down. TestRestTemplate was timing out before Mongo had a chance to timeout, and therefore throwing an exception rather than returning `503`. This, again, was due to a transitive dependency being out of date and overriding the more recent version. To fix this, we identified `api-sdk-manager-java-library` as the dependency that brought in the out of date transitive dep, so we removed this dependency entirely as it was not used in the service.

[DSND-2703](https://companieshouse.atlassian.net/browse/DSND-2703)

[DSND-2703]: https://companieshouse.atlassian.net/browse/DSND-2703?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/itest/java/uk/gov/companieshouse/pscdataapi/steps/PscDataSteps.java [hunk 1]
```diff
@@ -75,16 +95,16 @@ public class PscDataSteps {
...
 
     @Before
-    public void dbCleanUp(){
+    public void dbCleanUp() {
         if (!mongoDBContainer.isRunning()) {
             mongoDBContainer.start();
         }
         companyPscRepository.deleteAll();
-        MockitoAnnotations.initMocks(this);
+        MockitoAnnotations.openMocks(this);
```
Comment (johncook-ch) [inline src/itest/java/uk/gov/companieshouse/pscdataapi/steps/PscDataSteps.java:103]: This has new method returns an instance if `AutoCloseable`. According to the docs, the `@Before` method needs to be matched with an `@After` method that closes the object. e.g.
```
@BeforeEach public void openMocks() {
     closeable = MockitoAnnotations.openMocks(this);
}

@AfterEach public void releaseMocks() throws Exception {
    closeable.close();
} 
```
See [JavaDoc](https://javadoc.io/static/org.mockito/mockito-core/5.8.0/org/mockito/MockitoAnnotations.html) and [SO post](https://stackoverflow.com/questions/75970985/mockitoannotationns-openmocks-autoclosable)

**Review by johncook-ch — COMMENTED** [review]
Not yours, but there is an unused exception for method `aPSCExistsForForListSummary`  on line 1101

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Review by cgregoryCH — APPROVED** [review]

**Review by crowlandsCH — APPROVED** [review]
not on the scope of this ticket explicitly but there are a bunch of unused imports and exceptions we should clean up in future tickets

**Pull request merged**